### ISSUE TITLE
[2.4.0.RC2-1] "not found: value Route"

### ISSUE THREAD

**Issue by seratch** [scalatra/scalatra#525]
2.4.0.RC1 doesn't have this issue. This will effect existing applications.

- with 2.4.0.RC1 https://travis-ci.org/seratch/skinny-framework/builds/59010686 (also failure but no compilation errors)
- with 2.4.0.RC2-1 https://travis-ci.org/seratch/skinny-framework/builds/66810698

```
[error] /home/travis/build/seratch/skinny-framework/framework/src/main/scala/skinny/controller/feature/AngularXSRFProtectionFeature.scala:99: not found: value Route
[error]   before(isForged) {
[error]                    ^
[error] /home/travis/build/seratch/skinny-framework/framework/src/main/scala/skinny/controller/feature/BeforeAfterActionFeature.scala:84: not found: value Route
[error]   before() {
[error]            ^
[error] /home/travis/build/seratch/skinny-framework/framework/src/main/scala/skinny/controller/feature/BeforeAfterActionFeature.scala:88: not found: value Route
[error]   after() {
[error]           ^
[error] /home/travis/build/seratch/skinny-framework/framework/src/main/scala/skinny/controller/feature/RequestScopeFeature.scala:92: not found: value Route
[error]   before()(initializeRequestScopeAttributes(request))
[error]           ^
[error] /home/travis/build/seratch/skinny-framework/framework/src/main/scala/skinny/controller/feature/XContentTypeOptionsNosniffHeaderFeature.scala:17: not found: value Route
[error]   before() {
[error]            ^
[error] 5 errors found
[error] (framework/compile:compileIncremental) Compilation failed
```

**Comment by seratch** [comment]
TODO: I should create a minimum example to reproduce this issue in Scalatra project.

**Comment by xuwei-k** [comment]
maybe need fully-qualified class names

https://github.com/scalatra/scalatra/blob/v2.4.0.RC2-1/core/src/main/scala/org/scalatra/CoreDslMacros.scala#L78


bad

```scala
c.Expr[Unit](q"""routes.appendBeforeFilter(Route(Seq(..$transformers), () => $rescopedAction))""")
```

good

```scala
c.Expr[Unit](q"""routes.appendBeforeFilter(org.scalatra.Route(Seq(..$transformers), () => $rescopedAction))""")
```

**Comment by dozed** [comment]
This is fixed with #524
